src/App.tsx

```diff
@@ -1,24 +1,106 @@
-import React, { FC } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { FC, useState } from 'react';
+import {
+  Box,
+  Button,
+  Container,
+  CssBaseline,
+  TextField,
+} from '@material-ui/core';
+import Todo from './components/Todo/Todo';
+import { Todo as ITodo } from './components/Todo/typings';
 
-const App: FC = () => (
-  <div className="App">
-    <header className="App-header">
-      <img src={logo} className="App-logo" alt="logo" />
-      <p>
-        Edit <code>src/App.tsx</code> and save to reload.
-      </p>
-      <a
-        className="App-link"
-        href="https://reactjs.org"
-        target="_blank"
-        rel="noopener noreferrer"
+const App: FC = () => {
+  const [todoList, setTodoList] = useState<ITodo[]>([]);
+  const [newTodoName, setNewTodoName] = useState<string>('');
+  const [inputTodoError, setInputTodoError] = useState<string>('');
+
+  const changeTodoNameHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoName(event.target.value);
+  }
+
+  const addTodoHandler = () => {
+    const hasNewTodoAnInvalidName = !newTodoName || newTodoName.length < 6 || newTodoName.length > 20;
+    if (hasNewTodoAnInvalidName) {
+      setInputTodoError('Entrada inválida (digite de 6 a 20 caracteres)');
+      return;
+    } 
+    const newTodo: ITodo = {
+      title: newTodoName,
+      done: false,
+    };
+    const newTodoList = [newTodo, ...todoList];
+    setTodoList(newTodoList);
+  };
+
+  const changeTodoStatus = (position: number) => {
+    const updatedTodoList = todoList.map((todo, index) =>
+      index === position
+      ? Object.assign(todo, {done: !todo.done}) 
+      : todo);
+    setTodoList(updatedTodoList);
+  };
+
+  const removeTodoHandler = (position: number) => setTodoList(todoList.filter((_, index) => index !== position));
+  
+  return (
+    <>
+      <CssBaseline />
+      <Container
+        sx={{
+          height: '100vh',
+          display: 'flex',
+          flexDirection: 'column',
+        }}
       >
-        Learn React
-      </a>
-    </header>
-  </div>
-);
+        <Box
+          p={1}
+          sx={{
+            display: 'flex',
+          }}
+        >
+          <TextField
+            sx={{
+              mr: 'auto',
+            }}
+            error={!!inputTodoError}
+            value={newTodoName}
+            onChange={changeTodoNameHandler}
+            data-testid="add-task-input"
+            label="Nova Tarefa"
+            helperText={inputTodoError}
+            variant="filled"
+          />
+          <Button
+            onClick={addTodoHandler}
+            data-testid="add-task-btn"
+            variant="contained"
+            color="primary"
```

Comment:
When organizing props is very common to use it in this way:

1. Most important props like(id, data-testid, className)
2. Another props
3. callbacks

```suggestion
            data-testid="add-task-btn"
            color="primary"
            variant="contained"
            onClick={addTodoHandler}
```

Reply:
Resolved on ac40b7bd5b1418375c16876812c614ac7cb7f273